## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 452172`
Commit: [`ab100b917bfac4a8cdb371a38c671d60e699b93a`](https://github.com/NixOS/nixpkgs/commit/ab100b917bfac4a8cdb371a38c671d60e699b93a) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/ab100b917bfac4a8cdb371a38c671d60e699b93a..pull/452172/head))
Merge: [`740499b6d46348f715443745c50459627dd9a18e`](https://github.com/NixOS/nixpkgs/commit/740499b6d46348f715443745c50459627dd9a18e)

Logs: https://github.com/andrewzah/nixpkgs-review-gha/actions/runs/18631782950


---
### `x86_64-linux`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>nano-wallet</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>nano-wallet</summary>
<pre>[ 69%] Building CXX object submodules/rocksdb/CMakeFiles/rocksdb.dir/utilities/transactions/lock/range/range_tree/lib/locktree/keyrange.cc.o
[ 69%] Building CXX object submodules/rocksdb/CMakeFiles/rocksdb.dir/utilities/transactions/lock/range/range_tree/lib/locktree/lock_request.cc.o
[ 69%] Building CXX object submodules/rocksdb/CMakeFiles/rocksdb.dir/utilities/transactions/lock/range/range_tree/lib/locktree/locktree.cc.o
[ 69%] Building CXX object submodules/rocksdb/CMakeFiles/rocksdb.dir/utilities/transactions/lock/range/range_tree/lib/locktree/manager.cc.o
[ 69%] Building CXX object submodules/rocksdb/CMakeFiles/rocksdb.dir/utilities/transactions/lock/range/range_tree/lib/locktree/range_buffer.cc.o
[ 69%] Building CXX object submodules/rocksdb/CMakeFiles/rocksdb.dir/utilities/transactions/lock/range/range_tree/lib/locktree/treenode.cc.o
[ 69%] Building CXX object submodules/rocksdb/CMakeFiles/rocksdb.dir/utilities/transactions/lock/range/range_tree/lib/locktree/txnid_set.cc.o
[ 71%] Building CXX object submodules/rocksdb/CMakeFiles/rocksdb.dir/utilities/transactions/lock/range/range_tree/lib/locktree/wfg.cc.o
[ 71%] Building CXX object submodules/rocksdb/CMakeFiles/rocksdb.dir/utilities/transactions/lock/range/range_tree/lib/standalone_port.cc.o
[ 71%] Building CXX object submodules/rocksdb/CMakeFiles/rocksdb.dir/utilities/transactions/lock/range/range_tree/lib/util/dbt.cc.o
[ 71%] Building CXX object submodules/rocksdb/CMakeFiles/rocksdb.dir/utilities/transactions/lock/range/range_tree/lib/util/memarena.cc.o
[ 71%] Building CXX object submodules/rocksdb/CMakeFiles/rocksdb.dir/port/port_posix.cc.o
[ 71%] Building CXX object submodules/rocksdb/CMakeFiles/rocksdb.dir/env/env_posix.cc.o
[ 71%] Building CXX object submodules/rocksdb/CMakeFiles/rocksdb.dir/env/fs_posix.cc.o
[ 71%] Building CXX object submodules/rocksdb/CMakeFiles/rocksdb.dir/env/io_posix.cc.o
[ 71%] Building CXX object submodules/rocksdb/CMakeFiles/rocksdb.dir/__/__/build_version.cc.o
[ 71%] Linking CXX static library librocksdb.a
[ 71%] Built target rocksdb
make[1]: *** [CMakeFiles/Makefile2:2744: CMakeFiles/nano_wallet.dir/rule] Error 2
make: *** [Makefile:294: nano_wallet] Error 2</pre>
</details>
</details>

---
### `aarch64-linux`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>nano-wallet</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-linux`</summary>
<details>
<summary>nano-wallet</summary>
<pre>[ 70%] Building CXX object submodules/rocksdb/CMakeFiles/rocksdb.dir/utilities/transactions/lock/range/range_tree/lib/locktree/lock_request.cc.o
[ 70%] Building CXX object submodules/rocksdb/CMakeFiles/rocksdb.dir/utilities/transactions/lock/range/range_tree/lib/locktree/locktree.cc.o
[ 70%] Building CXX object submodules/rocksdb/CMakeFiles/rocksdb.dir/utilities/transactions/lock/range/range_tree/lib/locktree/manager.cc.o
[ 70%] Building CXX object submodules/rocksdb/CMakeFiles/rocksdb.dir/utilities/transactions/lock/range/range_tree/lib/locktree/range_buffer.cc.o
[ 70%] Building CXX object submodules/rocksdb/CMakeFiles/rocksdb.dir/utilities/transactions/lock/range/range_tree/lib/locktree/treenode.cc.o
[ 70%] Building CXX object submodules/rocksdb/CMakeFiles/rocksdb.dir/utilities/transactions/lock/range/range_tree/lib/locktree/txnid_set.cc.o
[ 70%] Building CXX object submodules/rocksdb/CMakeFiles/rocksdb.dir/utilities/transactions/lock/range/range_tree/lib/locktree/wfg.cc.o
[ 71%] Building CXX object submodules/rocksdb/CMakeFiles/rocksdb.dir/utilities/transactions/lock/range/range_tree/lib/standalone_port.cc.o
[ 71%] Building CXX object submodules/rocksdb/CMakeFiles/rocksdb.dir/utilities/transactions/lock/range/range_tree/lib/util/dbt.cc.o
[ 71%] Building CXX object submodules/rocksdb/CMakeFiles/rocksdb.dir/utilities/transactions/lock/range/range_tree/lib/util/memarena.cc.o
[ 71%] Building CXX object submodules/rocksdb/CMakeFiles/rocksdb.dir/util/crc32c_arm64.cc.o
[ 71%] Building CXX object submodules/rocksdb/CMakeFiles/rocksdb.dir/port/port_posix.cc.o
[ 71%] Building CXX object submodules/rocksdb/CMakeFiles/rocksdb.dir/env/env_posix.cc.o
[ 71%] Building CXX object submodules/rocksdb/CMakeFiles/rocksdb.dir/env/fs_posix.cc.o
[ 71%] Building CXX object submodules/rocksdb/CMakeFiles/rocksdb.dir/env/io_posix.cc.o
[ 71%] Building CXX object submodules/rocksdb/CMakeFiles/rocksdb.dir/__/__/build_version.cc.o
[ 71%] Linking CXX static library librocksdb.a
[ 71%] Built target rocksdb
make[1]: *** [CMakeFiles/Makefile2:2744: CMakeFiles/nano_wallet.dir/rule] Error 2
make: *** [Makefile:294: nano_wallet] Error 2</pre>
</details>
</details>

---
### `x86_64-darwin`
:white_check_mark: *No rebuilds*

---
### `aarch64-darwin`
:white_check_mark: *No rebuilds*

--- a/pkgs/applications/blockchains/nano-wallet/default.nix
+++ b/pkgs/applications/blockchains/nano-wallet/default.nix
@@ -15,24 +15,23 @@
 stdenv.mkDerivation rec {
 
   pname = "nano-wallet";
-  version = "25.1";
+  version = "28.2";
 
   src = fetchFromGitHub {
     owner = "nanocurrency";
     repo = "nano-node";
-    rev = "V${version}";
+    tag = "V${version}";
     fetchSubmodules = true;
-    hash = "sha256-YvYEXHC8kxviZLQwINs+pS61wITSfqfrrPmlR+zNRoE=";
+    hash = "sha256-Wo1Gd6dOnCoPiGmuJQhZmKKSg7LrKpfdvLNNKBYTUWI=";
   };
 
+  env.NIX_CFLAGS_COMPILE = "-Wno-error";
+
   patches = [
-    # Fix gcc-13 build failure due to missing <cstdint> includes.
+    # fix issue with <algorithm> include
     (fetchpatch {
-      name = "gcc-13.patch";
-      url = "https://github.com/facebook/rocksdb/commit/88edfbfb5e1cac228f7cc31fbec24bb637fe54b1.patch";
-      stripLen = 1;
-      extraPrefix = "submodules/rocksdb/";
-      hash = "sha256-HhlIYyPzIZFuyzHTUPz3bXgXiaFSQ8pVrLLMzegjTgE=";
+      url = "https://github.com/nanocurrency/nano-node/commit/1835a04dbbd1f6970649d7f72c454831432dd01f.patch";
+      hash = "sha256-IpC4yaIbJzQWYIC0QGXYQ345g6JnD2+xZG30qAQ1ubo=";
     })
   ];
 

